src/pdn/src/straps.cpp

```diff
@@ -404,12 +459,40 @@ void FollowPins::makeShapes(const Shape::ShapeTreeMap& other_shapes)
     auto* power_strap = new FollowPinShape(
         layer, power, odb::Rect(x0, power_y_bot, x1, power_y_bot + width));
     power_strap->addRow(row);
-    addShape(power_strap);
+    if (getExtendMode() == CLIP_CELLS) {
+      std::vector<Shape*> replacements;
+      if (power_strap->cut(power_obs_tree, grid, replacements)) {
+        for (Shape *cut_shape : replacements) {
+          if (cut_shape->getRect().dx() > 20 * width) {
+            cut_shape->setLocked();
+            addShape(cut_shape);
+          }
+        }
+      } else {
+        addShape(power_strap);
+      }
+    } else {
+      addShape(power_strap);
+    }
 
     auto* ground_strap = new FollowPinShape(
         layer, ground, odb::Rect(x0, ground_y_bot, x1, ground_y_bot + width));
     ground_strap->addRow(row);
-    addShape(ground_strap);
+    if (getExtendMode() == CLIP_CELLS) {
+      std::vector<Shape*> replacements;
+      if (ground_strap->cut(ground_obs_tree, grid, replacements)) {
+        for (Shape *cut_shape : replacements) {
+          if (cut_shape->getRect().dx() > 20 * width) {
+            cut_shape->setLocked();
```

Comment:
I think we need to dig into the issue that is forcing you to add this.
It might be helpful to have a testcase that demonstrate the issue so we can debug it.

Reply:
Hopefully this simplified illustration helps from a public github point of view - the detailed testcase is covered by NDA but could also be discussed further by email if necessary

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/78621419/13510ba2-8bfd-47cc-a189-fc06e4b9d7a0)

Reply:
since you will need to make a test for this PR, you can just use that test as the example of the bad behavior.